src/auth.js

```diff
@@ -63,110 +63,113 @@ export default class DropboxAuth {
+     * @returns {undefined}
+     */
   setClientSecret(clientSecret) {
     this.clientSecret = clientSecret;
   }
 
   /**
-   * Get the client secret
-   * @returns {String} Client secret
-   */
+     * Get the client secret
+     * @returns {String} Client secret
+     */
   getClientSecret() {
     return this.clientSecret;
   }
 
   /**
-   * Gets the refresh token
-   * @returns {String} Refresh token
-   */
+     * Gets the refresh token
+     * @returns {String} Refresh token
+     */
   getRefreshToken() {
     return this.refreshToken;
   }
 
   /**
-   * Sets the refresh token
-   * @param refreshToken - A refresh token
-   */
+     * Sets the refresh token
+     * @param refreshToken - A refresh token
+     */
   setRefreshToken(refreshToken) {
     this.refreshToken = refreshToken;
   }
 
   /**
-   * Gets the access token's expiration date
-   * @returns {Date} date of token expiration
-   */
+     * Gets the access token's expiration date
+     * @returns {Date} date of token expiration
+     */
   getAccessTokenExpiresAt() {
     return this.accessTokenExpiresAt;
   }
 
   /**
-   * Sets the access token's expiration date
-   * @param accessTokenExpiresAt - new expiration date
-   */
+     * Sets the access token's expiration date
+     * @param accessTokenExpiresAt - new expiration date
+     */
   setAccessTokenExpiresAt(accessTokenExpiresAt) {
     this.accessTokenExpiresAt = accessTokenExpiresAt;
   }
 
-  generatePKCECodes() {
-    let codeVerifier;
-    if (isBrowserEnv()) {
-      const array = new Uint8Array(PKCELength);
-      const randomValueArray = crypto.getRandomValues(array);
-      const base64String = btoa(randomValueArray);
-      codeVerifier = createBrowserSafeString(base64String).substr(0, 128);
-    } else {
-      const randomBytes = crypto.randomBytes(PKCELength);
-      codeVerifier = createBrowserSafeString(randomBytes).substr(0, 128);
-    }
+  /**
+     * Sets the code verifier for PKCE flow
+     * @param {String} codeVerifier - new code verifier
+     */
+  setCodeVerifier(codeVerifier) {
     this.codeVerifier = codeVerifier;
+  }
+
+  /**
+     * Gets the code verifier for PKCE flow
+     * @returns {String} - code verifier for PKCE
+     */
+  getCodeVerifier() {
+    return this.codeVerifier;
+  }
 
+  generateCodeChallenge() {
     const encoder = new Encoder();
-    const codeData = encoder.encode(codeVerifier);
+    const codeData = encoder.encode(this.codeVerifier);
     let codeChallenge;
     if (isBrowserEnv()) {
       return crypto.subtle.digest('SHA-256', codeData)
         .then((digestedHash) => {
-          const typedArray = new Uint8Array(digestedHash);
-          const base64String = btoa(typedArray);
-          codeChallenge = createBrowserSafeString(base64String).substr(0, 128);
+          const b64ed = btoa(String.fromCharCode.apply(null, new Uint8Array(digestedHash)));
+          codeChallenge = createBrowserSafeString(b64ed).substr(0, 128);
```

Comment:
Just for my understanding, why are we truncating the string?

Reply:
For the code challenge, spec says that we need to have a max of 128 characters so we substring to ensure that even post replacements and base 64 conversion it is still less than 128 characters